src/files.ts

```diff
@@ -79,24 +81,32 @@ export async function getProjectFiles(rootDir: string = path.join('.', '/'), cal
   // Note: filePaths contain relative paths such as "test/bar.ts", "../../src/foo.js"
   recursive(rootDir, async (err, filePaths) => {
     if (err) return callback(err, null, null);
+
     // Filter files that aren't allowed.
     const ignorePatterns = await DOTFILE.IGNORE();
-    filePaths.sort(); // Sort files alphanumerically
-    let abortPush = false;
-    const nonIgnoredFilePaths: string[] = [];
-    const file2path: Array<{ path: string; file: AppsScriptFile }> = [];  // used by `filePushOrder`
-    let ignoredFilePaths: string[] = [];
-    ignoredFilePaths = ignoredFilePaths.concat(ignorePatterns);
 
     // Replace OS specific path separator to common '/' char for console output
     filePaths = filePaths.map((name) => name.replace(/\\/g, '/'));
+    filePaths.sort(); // Sort files alphanumerically
 
     // check ignore files
-    const ignoreMatches = multimatch(filePaths, ignorePatterns, { dot: true });
-    const intersection: string[] = filePaths.filter(file => !ignoreMatches.includes(file));
+    const acc = filePaths.reduce(
```

Comment:
You can create ignoreMatches and intersection without using `reduce` .

For example ...

```
const ignoreMatches = [];
const intersection = [];
filePaths.foreach( file => {
  const relative = path.relative(rootDir, file);
  if (multimatch(relative, ignorePatterns, { dot: true }).length === 0) {
    intersection.push(file);
  } else {
    ignoreMatches.push(file);
  }
});
```

Reply:
Sure. I just like `.reduce()` too much.

Reply:
Don't use `reduce`, it's not readable.
Use a `forEach` like @takanakahiko's example.